src/renderers/dom/stack/client/ReactDOMComponent.js

```diff
@@ -253,6 +254,8 @@ if (__DEV__) {
     }
 
     validateDOMNesting(null, String(content), this, this._ancestorInfo);
+    validateDOMTag(this, this._ancestorInfo);
```

Comment:
I believe the changes in this file should be replicated in the Fiber component as well.